### PR TITLE
Derive OtlpMetricsExporter from MetricExporter

### PR DESCRIPTION
**Pull request by alanwest**

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ internal static class MetricItemExtensions
         internal static void AddBatch(
             this OtlpCollector.ExportMetricsServiceRequest request,
             OtlpResource.Resource processResource,
-            in Batch<Metric> batch)
+            in IEnumerable<Metric> batch)
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs:42]: Maybe change the variable name to `metrics` to align with the base class https://github.com/open-telemetry/opentelemetry-dotnet/blob/8e1a72b03cd1abf62493bafc589274c99870abf0/src/OpenTelemetry/Metrics/MetricExporter.cs#L29.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricsExporter.cs [hunk 2]
```diff
@@ -59,15 +72,33 @@ internal OtlpMetricsExporter(OtlpExporterOptions options, OtlpCollector.MetricsS
...
-        public override ExportResult Export(in Batch<Metric> batch)
+        public override ExportResult Export(IEnumerable<Metric> metrics)
         {
             // Prevents the exporter's gRPC and HTTP operations from being instrumented.
             using var scope = SuppressInstrumentationScope.Begin();
 
             var request = new OtlpCollector.ExportMetricsServiceRequest();
 
-            request.AddBatch(this.ProcessResource, batch);
+            request.AddBatch(this.ProcessResource, metrics);
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricsExporter.cs:101]:
```suggestion
            request.AddMetrics(this.ProcessResource, metrics);
```

**Review by reyang — APPROVED** [review]
LGTM.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricsExporter.cs (hunk 2 above)
Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricsExporter.cs:101]: #2313

**Pull request merged**